### PR TITLE
[WIP] Tomcat fix

### PR DESCRIPTION
**Pull request by joel-hamill**

### CODE REVIEW

File: 1.7/administration/locate-public-agent.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+- DC/OS is installed with at least 1 master and [public agent](/docs/1.7/overview/concepts/#public) node
+- DC/OS CLI is installed
+- [jQuery](https://github.com/stedolan/jq/wiki/Installation)
+- [SSH](/docs/1.7/administration/sshcluster/) configured
+
+You can find your public agent IP by running this command from the DC/OS CLI. This command SSHs to your cluster to obtain cluster information and then pings [ifconfig.co](https://ifconfig.co/) to determine your public IP address. 
+
+```
+$ echo "curl -s ifconfig.co" | dcos node ssh --master-proxy --mesos-id=$(dcos task --json | jq --raw-output '.[] | select(.name == "tomcat") | .slave_id') 2>/dev/null
```
Comment (sascala) [inline 1.7/administration/locate-public-agent.md]: @joel-hamill Is the `select(.name == "tomcat") part of this command necessary whenever a user wants to find their public agent IP?

**Comment by joel-hamill** [comment]
@sascala good catch, no it's not. @surdy what's the general command for finding public agents?

**Comment by mhausenblas** [comment]
@joel-hamill for example: `dcos node --json | jq "map(select(.attributes.public_ip).hostname)"`

**Comment by joel-hamill** [comment]
@mhausenblas that command displays the internal public IP.

**Comment by mhausenblas** [comment]
Yes, I know @joel-hamill. So your actual question was: how to find the Internet-routable, public IP address of a public agent? I think I've got it somewhere in https://github.com/mhausenblas/dcos-sd

**Comment by joel-hamill** [comment]
@mhausenblas part of the problem that @surdy and I have discovered is that [this](https://github.com/mhausenblas/dcos-sd#ip-address-of-node-a--marathon-app-is-running-on) doesn't seem to work unless you have already SSH'd to your node previously.

**Comment by mhausenblas** [comment]
It works otherwise I wouldn't have published it. See https://github.com/mhausenblas/dcos-sd#prerequisites

**Comment by surdy** [comment]
@joel-hamill if the host key verification problem had not happened this command would have given you the public IP of all the public agents on your cluster : 

`for id in $(dcos node --json | jq --raw-output '.[] | select(.reserved_resources.slave_public != null) | .id'); do echo "curl -s ifconfig.so" | dcos node ssh --master-proxy --mesos-id=$id ; done`

@mhausenblas I am running into `Host Key Verification failed` when I pipe commands into `dcos node ssh` unless I have sshed into the machine and exited before running the command. We can get it sometimes, but a partially working command in docs is a support nightmare. Hence I suggested not adding it in my conversation with @joel-hamill 

We can add functionality to dcos node to accept an optional command string argument to run the command string on the node : https://github.com/dcos/dcos-cli/issues/609

**Comment by joel-hamill** [comment]
@mhausenblas http://cl.ly/441K3K383a0t

**Comment by mhausenblas** [comment]
@joel-hamill that's because you executed the command on Mac OSX. Try it on Linux and tell me what you see ;)

**Comment by mhausenblas** [comment]
OK, whatever you guys think is best, from a UX and support perspective …

**Comment by sascala** [comment]
@joel-hamill is this PR blocked by the host key verification problem?

**Comment by joel-hamill** [comment]
@sascala yes, blocked

**Comment by sascala** [comment]
Closing until unblocked.

**Comment by surdy** [comment]
With DC/OS CLI 0.4.6 this works for getting Public IP of the node running tomcat 
```
dcos node ssh --option StrictHostKeyChecking=no --master-proxy --mesos-id=$(dcos task --json | jq --raw-output '.[] | select(.name == "tomcat") | .slave_id') "curl -s ifconfig.co" 2>/dev/null
```

The following gets the public IP of all Public Agents in the cluster
```
for id in $(dcos node --json | jq --raw-output '.[] | select(.reserved_resources.slave_public != null) | .id'); do dcos node ssh --master-proxy --mesos-id=$id "curl -s ifconfig.co" ; done 2>/dev/null
```

**Comment by joel-hamill** [comment]
@surdy is that the current version shipping with DC/OS, or?

**Comment by surdy** [comment]
0.4.6 is the current CLI version. CLI does not ship with DC/OS does it??

**Pull request closed without merging**